Move ADVANCED_PRIVACY_PROTECTIONS flags from WebKitAdditions and add Network.framework SPI

emw-apple opened the pull request:
#### 0728a2ae210e36af0b55f9fd8478eb99c56bfdf4
<pre>
Move ADVANCED_PRIVACY_PROTECTIONS flags from WebKitAdditions and add Network.framework SPI
<a href="https://bugs.webkit.org/show_bug.cgi?id=258429">https://bugs.webkit.org/show_bug.cgi?id=258429</a>

Reviewed by Wenson Hsieh.

Work towards making the build functional on macOS Sonoma. Publicizes the
flags used to control the web privacy features announced this year.

Reverts 261235@main (01d2717a458d6bfcc418c788caa2be8a57d25d8e). There is
no need to continue supporting older internal SDKs which don&apos;t have the
same function signature for nw_proxy_config_create_with_agent_data.

Disables AdvancedPrivacyProtections API tests which swizzle
WebPrivacy.framework on systems where the framework is not available.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm: Stop
  checking for private headers; CFNetworkSPI.h has all we need.
* Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm:
  Wrap QueryParameterRequestSwizzler and test cases that use it inside
  HAVE(SYSTEM_SUPPORT_FOR_ADVANCED_PRIVACY_PROTECTIONS).
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Proxy.mm: Stop checking for
  private headers; CFNetworkSPI.h has all we need.

Canonical link: <a href="https://commits.webkit.org/266123@main">https://commits.webkit.org/266123@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/65b12b510b1fc1291e6168141289f571f97c3186

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12929 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13253 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13583 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14667 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12339 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12997 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15757 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13275 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15036 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13095 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13808 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10946 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15118 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11096 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11700 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18757 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/11001 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12171 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11866 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15065 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/12238 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12325 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10221 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/12979 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11587 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3412 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3173 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15906 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/13351 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12166 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3204 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
```diff
@@ -96,6 +98,30 @@ typedef enum {
+#ifdef __cplusplus
+}
+#endif
+#endif // HAVE(NW_PROXY_CONFIG)
```
Let's make the `#endif` match the `#if`.

whsieh commented on Source/WTF/wtf/PlatformEnable.h:
```diff
@@ -1006,3 +1006,12 @@
 #endif
+
+#if !defined(ENABLE_ADVANCED_PRIVACY_PROTECTIONS) \
+    && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
```
Would this _disable_ advanced privacy protections on downlevels?

emw-apple replied:
I might be misunderstanding how this flag is supposed to work. HAVE_SYSTEM_SUPPORT_FOR_ADVANCED_PRIVACY_PROTECTIONS sounds obviously macOS 14+ only, but is this one supposed to match _any_ version of macOS we support? If so, maybe this can just be a `PLATFORM(MAC)`.

whsieh commented on Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm:
```diff
@@ -181,7 +181,16 @@ void waitUntilDidHandleRequest()
+    void SetUp() override
+    {
+        if (!PAL::isWebPrivacyFrameworkAvailable())
+            GTEST_SKIP() << "WebPrivacy.framework only available in iOS 16+, macOS 14+, STP, and downlevels";
```
Nit - `iOS 16+` -> `iOS 17+`.

whsieh commented on Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm:
```diff
@@ -849,7 +858,7 @@ HTTPServer server({
 }
 
-TEST(AdvancedPrivacyProtections, VerifyPixelsFromNoisyCanvas2DAPI)
+TEST_F(AdvancedPrivacyProtections, VerifyPixelsFromNoisyCanvas2DAPI)
```
I don't think these fingerprinting tests rely on WebPrivacy.framework. Can we keep these as `TEST(…)`?

whsieh reviewed (approved):
LGTM, with some comments; let's have @JonWBedard review the other gtest changes.

emw-apple committed:
Move ADVANCED_PRIVACY_PROTECTIONS flags from WebKitAdditions and add Network.framework SPI
https://bugs.webkit.org/show_bug.cgi?id=258429

Reviewed by Wenson Hsieh.

Work towards making the build functional on macOS Sonoma. Publicizes the
flags used to control the web privacy features announced this year.

Reverts 261235@main (01d2717a458d6bfcc418c788caa2be8a57d25d8e). There is
no need to continue supporting older internal SDKs which don't have the
same function signature for nw_proxy_config_create_with_agent_data.

Disables AdvancedPrivacyProtections API tests which swizzle
WebPrivacy.framework on systems where the framework is not available.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm: Stop
  checking for private headers; CFNetworkSPI.h has all we need.
* Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm:
  Wrap QueryParameterRequestSwizzler and test cases that use it inside
  HAVE(SYSTEM_SUPPORT_FOR_ADVANCED_PRIVACY_PROTECTIONS).
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Proxy.mm: Stop checking for
  private headers; CFNetworkSPI.h has all we need.

Canonical link: https://commits.webkit.org/266123@main
